### PR TITLE
Update sparseml to support torch 2.0

### PR DESCRIPTION
**Pull request by dsikka**
- Updated max version of `torch` to be 2.0, updated `torchvision` max to be `0.15.1`
- Updated max `torchaudio` to be `2.0.1`
- Tested and passed all `pytorch` tests 
- Tested updated package versions in training using `sparseml.image_classification.train`. Used the `ImageNette` dataset and `ResNet-50` model from SparseZoo

```
# Epoch and Learning-Rate variables
num_epochs: 3.0
init_lr: 0.0005

# quantization variables
quantization_epochs: 1.0

training_modifiers:
  - !EpochRangeModifier
    start_epoch: 0.0
    end_epoch: eval(num_epochs)

  - !LearningRateFunctionModifier
    final_lr: 0.0
    init_lr: eval(init_lr)
    lr_func: cosine
    start_epoch: 0.0
    end_epoch: eval(num_epochs)

# Phase 1 Sparse Transfer Learning / Recovery
sparse_transfer_learning_modifiers:
  - !ConstantPruningModifier
    start_epoch: 0.0
    params: __ALL_PRUNABLE__

# Phase 2 Apply quantization
sparse_quantized_transfer_learning_modifiers:
  - !QuantizationModifier
    start_epoch: eval(num_epochs - quantization_epochs)
```

- Trained using the sample recipe above and then exported in onnx using `sparseml.image_classification.export_onnx`
- Confirmed size of the onnx export model to match another model trained/exported using torch 1.13.1 (23 mb) and validated graph using Netron thanks to @KSGulin

### CODE REVIEW

**Review by bfineran — COMMENTED** [review]
thanks @dsikka - one last thing to look for is exporting quantized models with `convert_qat=True` and verifying that the weights are properly compressed (ie load a quantized resnet pytorch model from the sparsezoo, export it to onnx and verify that the file size is the expected quantized one)

**Comment by dbogunowicz** [comment]
Oh, wow, this seems to be much less effort than expected no?

**Review by bfineran — DISMISSED** [review]

File: src/sparseml/pytorch/base.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
 
 _TORCH_MIN_VERSION = "1.0.0"
-_TORCH_MAX_VERSION = "1.13.100"  # set bug to 100 to support all future 1.9.X versions
+_TORCH_MAX_VERSION = "2.0.0"  # set bug to 100 to support all future 1.9.X versions
```
Comment (rahul-tuli) [inline src/sparseml/pytorch/base.py:52]: Could we remove this outdated comment?

**Review by rahul-tuli — COMMENTED** [review]
LGTM after outdated comment removal!

File: src/sparseml/pytorch/base.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
 
 
 _TORCH_MIN_VERSION = "1.0.0"
-_TORCH_MAX_VERSION = "1.13.100"  # set bug to 100 to support all future 1.9.X versions
+_TORCH_MAX_VERSION = "2.0.0"
```
Comment (bfineran) [inline src/sparseml/pytorch/base.py:52]: Actually building on Rahul's comment, we should match the patch version to .100 still to support future patches of 2.0.X

**Review by bfineran — CHANGES_REQUESTED** [review]

File: src/sparseml/pytorch/base.py (hunk 2 above)
Comment (dsikka) [reply src/sparseml/pytorch/base.py:52]: I.e set `2.0.100` so that we can support all `2.0.x` versions in the future? @bfineran

**Review by bfineran — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**